.github/workflows/ci.yml

```diff
@@ -1,16 +1,98 @@
+name: build
+
 on: [push, pull_request]
 
 jobs:
   android-bin:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - run: bazel build -c opt :encoder_main --config=android_arm64
-      - run: bazel build -c opt :decoder_main --config=android_arm64
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Build encoder_main
+        run: |
+          export ANDROID_NDK_HOME="$ANDROID_HOME/ndk/21.4.7075529"
+          bazel build -c opt :encoder_main --config=android_arm64
+      - name: Build decoder_main
+        run: |
+          export ANDROID_NDK_HOME="$ANDROID_HOME/ndk/21.4.7075529"
+          bazel build -c opt :decoder_main --config=android_arm64
+      - name: Copy artifacts
+        run: |
+          mkdir action-product
+          cp -r wavegru action-product/
+          cp bazel-bin/encoder_main action-product/lyra-encoder
+          cp bazel-bin/decoder_main action-product/lyra-decoder
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: lyra-android-bin
+          path: action-product
 
   android-app:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Build Android app
-        run: bazel build android_example:lyra_android_example --config=android_arm64 --copt=-DBENCHMARK
+      - name: Checkout repo
```

Comment:
Thanks, it's useful to have these targets.  However, there are many steps that are duplicated across these.  Could you refactor as much as possible below this line with action composition?

https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/

Reply:
Yes, I just saw that PR being merged, and I've already resolved the conflict now. This PR contains a change that is exactly equivalent to #15.

Reply:
Great.  Please take a look at the refactoring comment.

Reply:
Hi, just checking in on this.  Would you be able to address the refactoring comment?

Reply:
Hi, I just want to note that I will be taking up this PR and will add the changes I mentioned myself.  If you prefer to do them yourself, please let me know.

Reply:
I'm so sorry I'm a little busy at this time so I didn't pay attention to it. But I'm happy to do that if you haven't started yet!